File: Array/wuchao/array/8-7 判断数组是否存在重复元素/ContainsDuplicate.java
package com.slow.lib.practice.others;

import java.util.HashMap;

/**
 * @Author wuchao
 * @Date 2020/8/7-9:40 AM
 * @description
 * @email 329187218@qq.com
 * @see
 */
public class ContainsDuplicate {
    public boolean containsDuplicate(int[] nums) {
        HashMap<Integer,Object> map = new HashMap();
        final Object dummy = new Object();
        for(int item:nums){
            if(map.put(item,dummy)!=null) return true;
        }
        return false;
    }
}

File: Array/wuchao/array/8-7 判断数组是否存在重复元素/ContainsDuplicate.kt
package com.slow.lib.practice.others

import java.util.*

/**
 * @Author wuchao
 * @Date 2020/8/7-9:40 AM
 * @see
 * @description
 * @email 329187218@qq.com
 */
class ContainsDuplicate {
    fun containsDuplicate(nums: IntArray): Boolean {
        val hashSet = HashSet<Int>()
        for(item in nums){
            if(!hashSet.add(item)) return true
        }
        return false
    }
}